icu4c/source/common/stringtriebuilder.cpp

```diff
@@ -404,7 +404,7 @@ StringTrieBuilder::FinalValueNode::operator==(const Node &other) const {
     if(!Node::operator==(other)) {
         return false;
     }
-    const FinalValueNode &o=(const FinalValueNode &)other;
+    const FinalValueNode &o=dynamic_cast<const FinalValueNode &>(other);
```

Comment:
I am sure these can all be static_cast's, because otherwise the code would not work. Besides, we must not throw exceptions (see above).

For example, in this operator==, the call to Node::operator== a few lines up ensures that `other` has the same type as this class.

Reply:
ok, I see. that is not obvious to me such line will "ensures that other has the same type as this class."
Changing the dynamic_cast to static_cast will not cause run time error under CFI so I will change them.